client/ts/src/connection/checker.ts

```diff
@@ -77,11 +88,54 @@ export class Checker {
   async check(): Promise<State> {
     const prevStatus = this._state.status;
     try {
-      const [res, err] = await this.client.send(Checker.ENDPOINT, {}, z.object({}), responseZ);
+      const [res, err] = await this.client.send(
+        Checker.ENDPOINT,
+        {},
+        z.object({}),
+        responseZ,
+      );
       if (err != null) throw err;
+      if (res.nodeVersion == null) {
+        this._state.clientServerCompatible = false;
+        if (!this.versionWarned)
+          console.warn(`
+          Synnax cluster node version is too old for client version ${this.clientVersion}.
```

Comment:
Can you consolidate these error statements into one statement somewhere that accepts arguments for old / new and the versions? It's pretty hard to follow the logic of so many nested if-else statements and that would make it easier.